explore/src/main/scala/explore/targeteditor/AladinCell.scala

```diff
@@ -76,28 +75,24 @@ object AladinCell extends ModelOptics {
       // flag to trigger centering. This is a bit brute force but
       // avoids us needing a ref to a Fn component
       .useStateViewWithReuse(false)
-      // GuideStar candidates (loading, list)
-      .useState((true, List.empty[GuideStarCandidate]))
-      // Lisen on web worker for messages with catalog candidates
-      .useEffectOnMountBy((props, _, _, _, gs) =>
-        props.ctx.worker.stream
-          .evalMap(m =>
-            decodeFromTransferable[IO, CatalogResults](m)(gsl =>
-              gs.setState((false, gsl.candidates)).to[IO]
+      // Listen on web worker for messages with catalog candidates
+      .useStreamBy((props, _, _, _) => (props.target.get, props.obsConf))((props, _, _, _) =>
+        deps =>
+          props.ctx.worker.stream
+            .map(decodeFromTransferable[CatalogResults])
+            .unNone
+            .map(_.candidates)
+            .merge(
```

Comment:
Looks longger than the original

Reply:
Well, it was 20 lines before and now it's 16.